src/loginAzurePowerShell.ts

```diff
@@ -0,0 +1,76 @@
+import * as os from 'os';
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+
+var psPath: string;
+
+export const initializeAz = async (servicePrincipalId: string, servicePrincipalKey: string, tenantId: string, subscriptionId: string) => {
+    psPath = await io.which("pwsh", true);
+    await importModule();
+    await loginToAzure(servicePrincipalId, servicePrincipalKey, tenantId, subscriptionId);
+}
+
+async function importModule() {
+    setPSModulePath();
+    const prefix = "az_";
+    const moduleName: string = "Az.Accounts";
+    const azLatestVersion: string = await getLatestModule(moduleName);
+    core.debug(`Az Module version used: ${azLatestVersion}`);
+    setPSModulePath(`${prefix}${azLatestVersion}`);
+}
+
+function setPSModulePath(azPSVersion: string = "") {
+    let modulePath: string = "";
+    const RUNNER: string = process.env.RUNNER_OS || os.type();
+    switch (RUNNER) {
+        case "Linux":
+            modulePath = `/usr/share/${azPSVersion}:`;
+            break;
+        case "Windows":
+        case "Windows_NT":
+            modulePath = `C:\\Modules\\${azPSVersion};`;
+            break;
+        case "macOS":
+        case "Darwin":
+            // TODO: add modulepath
+            break;
+    }
+    process.env.PSModulePath = `${modulePath}${process.env.PSModulePath}`;
```

Comment:
process.env.PSModulePath will not be available in javascript process.

Reply:
it is available, i have tested